### PR TITLE
Fix Clang-Tidy warnings

### PR DESCRIPTION
**Pull request by AtariDreams**
This is a heavily stripped down version of the PR I sent over a year ago. Hopefully this is much better.

### CODE REVIEW

**Comment by AtariDreams** [comment]
All done and all ready!

**Comment by AtariDreams** [comment]
Ready for review!

File: cups/debug.c [hunk 1]
```diff
@@ -616,7 +616,7 @@ _cups_safe_vsnprintf(
 
 	        *bufptr++ = '\\';
 		*bufptr++ = '0';
-		*bufptr++ = '0' + *s / 8;
+		*bufptr++ = '0' + (*s >> 3);
```
Comment (michaelrsweet) [inline cups/debug.c:619]: Really no benefit to this change and "/ 8" is more obvious than ">> 3".

File: cups/http-addrlist.c [hunk 2]
```diff
@@ -234,8 +236,9 @@ httpAddrConnect2(
       if (fds[nfds] > max_fd)
 	max_fd = fds[nfds];
 #endif /* !HAVE_POLL */
-
+#ifdef O_NONBLOCK
```
Comment (michaelrsweet) [inline cups/http-addrlist.c:238]: Addresses still need to be set, so this conditional stuff isn't correct.

File: cups/ipp-support.c [hunk 3]
```diff
@@ -740,9 +740,7 @@ ippAttributeString(
 
       case IPP_TAG_DATE :
           {
-            unsigned year;		/* Year */
-
-            year = ((unsigned)val->date[0] << 8) + (unsigned)val->date[1];
+            unsigned year = ((unsigned)val->date[0] << 8) | (unsigned)val->date[1];
```
Comment (michaelrsweet) [inline cups/ipp-support.c:743]: Still need a comment for the variable...

File: cups/ipp.c [hunk 4]
```diff
@@ -2910,9 +2910,8 @@ ippReadIO(void       *src,		/* I - Data source */
 
           ipp->request.any.version[0]  = buffer[0];
           ipp->request.any.version[1]  = buffer[1];
-          ipp->request.any.op_status   = (buffer[2] << 8) | buffer[3];
-          ipp->request.any.request_id  = (((((buffer[4] << 8) | buffer[5]) << 8) |
-	                        	 buffer[6]) << 8) | buffer[7];
+          ipp->request.any.op_status = (buffer[2] << 8) | buffer[3];
```
Comment (michaelrsweet) [inline cups/ipp.c:2913]: Align "="

File: cups/ipp.c [hunk 5]
```diff
@@ -3363,14 +3360,9 @@ ippReadIO(void       *src,		/* I - Data source */
...
 
-                value->resolution.xres =
-		    (((((buffer[0] << 8) | buffer[1]) << 8) | buffer[2]) << 8) |
-		    buffer[3];
-                value->resolution.yres =
-		    (((((buffer[4] << 8) | buffer[5]) << 8) | buffer[6]) << 8) |
-		    buffer[7];
-                value->resolution.units =
-		    (ipp_res_t)buffer[8];
+                value->resolution.xres = (buffer[0] << 24) | (buffer[1] << 16) | (buffer[2] << 8) | buffer[3];
```
Comment (michaelrsweet) [inline cups/ipp.c:3363]: Align "=".

File: cups/ipp.c [hunk 6]
```diff
@@ -4701,8 +4689,8 @@ ippValidateAttribute(
 		userpass[256],		/* Username/password from URI */
 		hostname[256],		/* Hostname from URI */
 		resource[1024];		/* Resource from URI */
-  int		port,			/* Port number from URI */
-		uri_status;		/* URI separation status */
+  int		port;			/* Port number from URI */
+  http_uri_status_t		uri_status;		/* URI separation status */
```
Comment (michaelrsweet) [inline cups/ipp.c:4693]: Fix indentation.

File: cups/raster-interpret.c [hunk 7]
```diff
@@ -712,26 +712,26 @@ cleartomark_stack(_cups_ps_stack_t *st)	/* I - Stack */
 
 static int				/* O - 0 on success, -1 on error */
 copy_stack(_cups_ps_stack_t *st,	/* I - Stack */
-           int              c)		/* I - Number of objects to copy */
+           int              count)		/* I - Number of objects to copy */
```
Comment (michaelrsweet) [inline cups/raster-interpret.c:715]: Fix comment indentation, although I'm not sure it is worthwhile to change this code that is going away.

File: cups/request.c [hunk 8]
```diff
@@ -102,7 +102,7 @@ cupsDoIORequest(http_t     *http,	/* I - Connection to server or @code CUPS_HTTP
 		int        outfile)	/* I - File to write to or -1 for none */
 {
   ipp_t		*response = NULL;	/* IPP response data */
-  size_t	length = 0;		/* Content-Length value */
+  size_t	length;		/* Content-Length value */
```
Comment (michaelrsweet) [inline cups/request.c:105]: Fix comment indentation

File: cups/tls-darwin.c [hunk 9]
```diff
@@ -84,8 +84,8 @@ cupsMakeServerCredentials(
 {
 #if TARGET_OS_OSX
   int		pid,			/* Process ID of command */
-		status,			/* Status of command */
-		i;			/* Looping var */
+		status;			/* Status of command */
+	size_t		i;			/* Looping var */
```
Comment (michaelrsweet) [inline cups/tls-darwin.c:88]: Fix indentation.

File: cups/tls-darwin.c [hunk 10]
```diff
@@ -1667,6 +1666,8 @@ _httpTLSStart(http_t *http)		/* I - HTTP connection */
 	      {
 		if ((names = cupsArrayNew(NULL, NULL)) != NULL)
 		{
+      CFIndex count; /* Number of credentials */
```
Comment (michaelrsweet) [inline cups/tls-darwin.c:1669]: Fix indentation

File: cups/tls-darwin.c [hunk 11]
```diff
@@ -1801,7 +1802,7 @@ _httpTLSWrite(http_t     *http,		/* I - HTTP connection */
 	       const char *buf,		/* I - Buffer holding data */
 	       int        len)		/* I - Length of buffer */
 {
-  ssize_t	result;			/* Return value */
+  int	result;			/* Return value */
```
Comment (michaelrsweet) [inline cups/tls-darwin.c:1805]: Fix indentation.

File: cups/tls-gnutls.c [hunk 12]
```diff
@@ -1120,7 +1120,7 @@ http_gnutls_read(
   ssize_t	bytes;			/* Bytes read */
 
 
-  DEBUG_printf(("5http_gnutls_read(ptr=%p, data=%p, length=%d)", ptr, data, (int)length));
+  DEBUG_printf(("5http_gnutls_read(ptr=%p, data=%p, length=%lu)", ptr, data, length));
```
Comment (michaelrsweet) [inline cups/tls-gnutls.c:1123]: "%lu" is not the format specifier for `size_t`.

File: filter/rastertoepson.c [hunk 13]
```diff
@@ -877,7 +876,7 @@ OutputRows(
 
     n = dot_count / DotBytes;
     putchar((int)(n & 255));
-    putchar((int)(n / 256));
+    putchar((int)(n >> 8));
```
Comment (michaelrsweet) [inline filter/rastertoepson.c:879]: Unnecessary change.

File: filter/rastertoepson.c [hunk 14]
```diff
@@ -920,9 +919,9 @@ OutputRows(
 
       n = (unsigned)dot_count / DotBytes;
       putchar((int)(n & 255));
-      putchar((int)(n / 256));
+      putchar((int)(n >> 8));
```
Comment (michaelrsweet) [inline filter/rastertoepson.c:922]: Unnecessary change.

File: scheduler/client.c [hunk 15]
```diff
@@ -553,7 +553,7 @@ cupsdReadClient(cupsd_client_t *con)	/* I - Client to read from */
 			*ptr;		/* Pointer into strings */
   http_status_t		status;		/* Transfer status */
   ipp_state_t		ipp_state;	/* State of IPP transfer */
-  int			bytes;		/* Number of bytes to POST */
+  ssize_t			bytes;		/* Number of bytes to POST */
```
Comment (michaelrsweet) [inline scheduler/client.c:556]: Fix indentation.

File: scheduler/client.c [hunk 16]
```diff
@@ -1542,7 +1542,7 @@ cupsdReadClient(cupsd_client_t *con)	/* I - Client to read from */
             if (write(con->file, line, (size_t)bytes) < bytes)
 	    {
               cupsdLogClient(con, CUPSD_LOG_ERROR,
-	                     "Unable to write %d bytes to \"%s\": %s", bytes,
+	                     "Unable to write %zi bytes to \"%s\": %s", bytes,
```
Comment (michaelrsweet) [inline scheduler/client.c:1545]: "%zi" is not portable.

File: scheduler/client.c [hunk 17]
```diff
@@ -1722,7 +1722,7 @@ cupsdReadClient(cupsd_client_t *con)	/* I - Client to read from */
               if (write(con->file, line, (size_t)bytes) < bytes)
 	      {
         	cupsdLogClient(con, CUPSD_LOG_ERROR,
-	                       "Unable to write %d bytes to \"%s\": %s",
+	                       "Unable to write %zi bytes to \"%s\": %s",
```
Comment (michaelrsweet) [inline scheduler/client.c:1725]: "%zi" is not portable.

File: scheduler/client.c [hunk 18]
```diff
@@ -2244,8 +2244,8 @@ cupsdUpdateCGI(void)
 void
 cupsdWriteClient(cupsd_client_t *con)	/* I - Client connection */
 {
-  int		bytes,			/* Number of bytes written */
-		field_col;		/* Current column */
+  ssize_t		bytes;			/* Number of bytes written */
+	int		field_col;		/* Current column */
```
Comment (michaelrsweet) [inline scheduler/client.c:2248]: Fix indentation.

File: scheduler/client.c [hunk 19]
```diff
@@ -2736,14 +2736,14 @@ get_file(cupsd_client_t *con,		/* I  - Client connection */
 	  snprintf(filename, len, "%s/ppd/%s.ppd", ServerRoot, p->printers[i]->name);
 	  if (!access(filename, 0))
 	  {
-	    p = p->printers[i];
+	    // p = p->printers[i];
```
Comment (michaelrsweet) [inline scheduler/client.c:2739]: No.

File: scheduler/cupsfilter.c [hunk 20]
```diff
@@ -856,7 +856,7 @@ exec_filter(const char *filter,		/* I - Filter to execute */
 #if defined(__APPLE__)
   char		processPath[1024],	/* CFProcessPath environment variable */
 		linkpath[1024];		/* Link path for symlinks... */
-  int		linkbytes;		/* Bytes for link path */
+  ssize_t		linkbytes;		/* Bytes for link path */
```
Comment (michaelrsweet) [inline scheduler/cupsfilter.c:859]: Fix indentation.

File: scheduler/env.c [hunk 21]
```diff
@@ -261,8 +261,9 @@ find_env(const char *name)		/* I - Variable name */
   int		i;			/* Looping var */
   size_t	namelen;		/* Length of name */
 
+  namelen = strlen(name);
```
Comment (michaelrsweet) [inline scheduler/env.c:264]: Two blank lines after declarations at the first level under a function. But why move the initializer out of the for loop?

File: scheduler/policy.c [hunk 22]
```diff
@@ -268,6 +268,7 @@ cupsdGetPrivateAttrs(
 
 
 #ifdef DEBUG
+  assert(policy != NULL);
```
Comment (michaelrsweet) [inline scheduler/policy.c:271]: NEVER NEVER NEVER use assert in CUPS.

File: scheduler/process.c [hunk 23]
```diff
@@ -491,7 +491,7 @@ cupsdStartProcess(
 #if defined(__APPLE__)
   char		processPath[1024],	/* CFProcessPath environment variable */
 		linkpath[1024];		/* Link path for symlinks... */
-  int		linkbytes;		/* Bytes for link path */
+  ssize_t		linkbytes;		/* Bytes for link path */
```
Comment (michaelrsweet) [inline scheduler/process.c:494]: Fix indentation.

File: scheduler/util.c [hunk 24]
```diff
@@ -161,11 +161,11 @@ cupsdExec(const char *command,		/* I - Full path to program */
           char       **argv)		/* I - Command-line arguments */
 {
 #ifdef __APPLE__
-  int	i, j;				/* Looping vars */
+  size_t	i, j;				/* Looping vars */
```
Comment (michaelrsweet) [inline scheduler/util.c:164]: Fix indentation.

File: systemv/cupstestppd.c [hunk 25]
```diff
@@ -3343,10 +3343,10 @@ check_sizes(ppd_file_t *ppd,		/* I - PPD file */
       }
       else
       {
-        width_tmp  = (fabs(size->width - ceil(size->width)) < 0.1) ?
-	                 ceil(size->width) : size->width;
-        length_tmp = (fabs(size->length - ceil(size->length)) < 0.1) ?
-	                 ceil(size->length) : size->length;
+        width_tmp  = (fabs(size->width - ceilf(size->width)) < 0.1f) ?
```
Comment (michaelrsweet) [inline systemv/cupstestppd.c:3346]: ceilf is not portable.

File: tools/ippeveprinter.c [hunk 26]
```diff
@@ -403,7 +403,7 @@ main(int  argc,				/* I - Number of command-line args */
 		ppm = 10,		/* Pages per minute for mono */
 		ppm_color = 0,		/* Pages per minute for color */
 		web_forms = 1;		/* Enable web site forms? */
-  ipp_t		*attrs = NULL;		/* Printer attributes */
+  ipp_t		*attrs;		/* Printer attributes */
```
Comment (michaelrsweet) [inline tools/ippeveprinter.c:406]: Fix indentation.

File: tools/ippeveprinter.c [hunk 27]
```diff
@@ -6637,8 +6637,8 @@ process_job(ippeve_job_t *job)		/* I - Job */
     * Execute a command with the job spool file and wait for it to complete...
     */
 
-    int 		pid,		/* Process ID */
-			status;		/* Exit status */
+
+	int		status;		/* Exit status */
```
Comment (michaelrsweet) [inline tools/ippeveprinter.c:6641]: Fix indentation, whitespace.

File: tools/ippeveps.c [hunk 28]
```diff
@@ -533,8 +532,8 @@ jpeg_to_ps(const char    *filename,	/* I - Filename */
   int		copy;			/* Current copy */
   int		width = 0,		/* Width */
 		height = 0,		/* Height */
-		depth = 0,		/* Number of colors */
-		length;			/* Length of marker */
+		depth = 0;		/* Number of colors */
+	size_t		length;			/* Length of marker */
```
Comment (michaelrsweet) [inline tools/ippeveps.c:536]: Fix indentation.

File: tools/ippfind.c [hunk 29]
```diff
@@ -335,21 +335,23 @@ main(int  argc,				/* I - Number of command-line args */
           if ((temp = new_expr(IPPFIND_OP_EXEC, invert, NULL, NULL,
                                argv + i)) == NULL)
             return (IPPFIND_EXIT_MEMORY);
-
-          while (i < argc)
+          
+          do
```
Comment (michaelrsweet) [inline tools/ippfind.c:339]: Braces around do...while

File: tools/ippfind.c [hunk 30]
```diff
@@ -1150,7 +1152,7 @@ main(int  argc,				/* I - Number of command-line args */
 
   avahi_client = avahi_client_new(avahi_simple_poll_get(avahi_poll),
 			          0, client_callback, avahi_poll, &err);
-  if (!avahi_client)
+  if (!avahi_client && err)
   {
```
Comment (michaelrsweet) [inline tools/ippfind.c:1156]: NULL avahi_client means no Bonjour/DNS-SD. If Avahi doesn't return an error on an error it is their fuckup.

File: tools/printer-lg-png.h [hunk 31]
```diff
@@ -1845,6 +1845,6 @@ static const unsigned char printer_lg_png[] =
   0x54,0xbd,0x25,0xba,0x49,0x71,0x9e,0x59,0xba,0x4d,0x53,0x1f,0xe6,0xed,0x4b,0xb9,
   0xde,0x17,0x45,0xf8,0x6b,0x81,0xbf,0xa8,0x8c,0x3b,0xff,0xff,0x00,0xa9,0xd9,0x66,
   0x33,0x79,0x3c,0x5d,0x0d,0x00,0x00,0x00,0x00,0x49,0x45,0x4e,0x44,0xae,0x42,0x60,
-  0x82,
+  0x82
```
Comment (michaelrsweet) [inline tools/printer-lg-png.h:1848]: These are auto-generated and will get changed whenever the source PNG image is changed.

File: vcnet/regex/debug.c [hunk 32]
```diff
@@ -15,14 +15,12 @@
  == void regprint(regex_t *r, FILE *d);
  */
 void
-regprint(r, d)
-regex_t *r;
-FILE *d;
+regprint(regex_t *r, FILE *d)
```
Comment (michaelrsweet) [inline vcnet/regex/debug.c:18]: The regex code in the vcnet directory is not maintained by the CUPS project. No changed will be accepted here.

**Review by michaelrsweet — CHANGES_REQUESTED** [review]
There are a lot of unrelated changes in one PR here, which should be broken out. Watch for GCC extensions! Also, the regex library changes for Windows won't be accepted because we pull it from the upstream project.

File: backend/ieee1284.c [hunk 33]
```diff
@@ -46,7 +46,7 @@ backendGetDeviceID(
 
 #else /* Get the device ID from the specified file descriptor... */
 #  ifdef __linux
-  int	length;				/* Length of device ID info */
+  ssize_t	length;				/* Length of device ID info */
```
[the remark this answers is not in the document]
Comment (AtariDreams) [reply backend/ieee1284.c:49]: > 

Where is the upstream for that then?

**Comment by michaelrsweet** [comment]
Upstream for Henry Spencer's regex library appears to be [here](https://github.com/garyhouston/regex).

File: backend/testbackend.c [hunk 34]
```diff
@@ -561,12 +566,12 @@ main(int  argc,				/* I - Number of command-line args */
       length   = sizeof(buffer);
       scstatus = cupsSideChannelSNMPGet(oid, buffer, &length, 5.0);
       printf("CUPS_SC_CMD_SNMP_GET %s returned %s, %d bytes (%s)\n", oid,
-	     statuses[scstatus], (int)length, buffer);
+	     statuses[scstatus], length, buffer);
```
[the remark this answers is not in the document]
Comment (AtariDreams) [reply backend/testbackend.c:569]: Length is an int already in this case though.

[the remark this answers is not in the document]
Comment (AtariDreams) [reply backend/testbackend.c:569]: The cast is redundant

File: cups/http-addrlist.c (hunk 2 above)
Comment (AtariDreams) [reply cups/http-addrlist.c:239]: It is not read though, so what is the point?

Comment (AtariDreams) [reply cups/http-addrlist.c:239]: addrs is not read or even used

File: tools/ippeveps.c (hunk 28 above)
Comment (AtariDreams) [reply tools/ippeveps.c:536]: The indentation is the same.

File: cups/ipp-support.c (hunk 3 above)
Comment (AtariDreams) [reply cups/ipp-support.c:743]: year is pretty self explanatory though.

File: cups/snmp.c [hunk 35]
```diff
@@ -93,7 +93,7 @@ _cupsSNMPCopyOID(int       *dst,	/* I - Destination OID */
   DEBUG_printf(("4_cupsSNMPCopyOID(dst=%p, src=%p, dstsize=%d)", dst, src,
                 dstsize));
 
-  for (i = 0, dstsize --; src[i] >= 0 && i < dstsize; i ++)
+  for (i = 0, dstsize --; i < dstsize && src[i] >= 0; i ++)
```
[the remark this answers is not in the document]
Comment (AtariDreams) [reply cups/snmp.c:96]: Codacy says otherwise.

File: cups/ipp.c (hunk 6 above)
Comment (AtariDreams) [reply cups/ipp.c:4693]: Indentation matches the other text, even if it is rendered funny in GitHub

File: systemv/cupstestppd.c (hunk 25 above)
Comment (AtariDreams) [reply systemv/cupstestppd.c:3346]: ceilf has been part of the standard since c99.

Comment (AtariDreams) [reply systemv/cupstestppd.c:3346]: Wait, is this supposed to be ANSI C?

**Comment by AtariDreams** [comment]
Done! @michaelrsweet

File: cups/http-addrlist.c (hunk 2 above)
Comment (AtariDreams) [reply cups/http-addrlist.c:239]: Something here is definitely not working as intended if this is the case

Comment (AtariDreams) [reply cups/http-addrlist.c:239]: Only the code under ifdef O_block is reading from the addrlist array, so if not assigning to that variable is not correct, neither is the current code.

**Comment by AtariDreams** [comment]
@michaelrsweet Every requested change has been made!

**Review by michaelrsweet — CHANGES_REQUESTED** [review]
You keep adding things, and this PR is getting huge.

I just want to see genuine bug fixes here, no code style/pattern changes. Anything more advanced than that should be applied against one of the CUPS 3.x repositories.

**Comment by AtariDreams** [comment]
I am not adding anything else here.

**Comment by AtariDreams** [comment]
No more changes outside of requested changes and rollbacks now! @michaelrsweet

**Comment by michaelrsweet** [comment]
OK, this is just too much in one PR to review and seems to be more than "clang tidy" changes...
